Sources/Kommunicate/Classes/Kommunicate.swift

```diff
@@ -433,22 +435,127 @@ open class Kommunicate: NSObject,Localizable{
         }
     }
 
-    open class func isPreChatEnabled(completion: @escaping(Bool?) -> ()) {
+    open class func createConversationWithPreChatWith(appID: String, conversation: KMConversation?, viewController: UIViewController, completion: @escaping(KommunicateError?) -> ()) {
+        
+        KMUserDefaultHandler.setApplicationKey(appID)
+        Kommunicate.presentingViewController = viewController
+    
         let kmAppSetting = KMAppSettingService()
         kmAppSetting.appSetting { (result) in
             switch result {
             case .success(let appSetting):
                 guard let isPreChatEnable = appSetting.collectLead else { return }
                 if isPreChatEnable {
-                    completion(true)
+                    if !KMUserDefaultHandler.isLoggedIn() {
+                        DispatchQueue.main.async {
+                            let preChatVC = KMPreChatFormViewController(configuration: Kommunicate.defaultConfiguration)
+                            preChatVC.delegate = Kommunicate()
+                            viewController.present(preChatVC, animated: false, completion: nil)
+                        }
+                    }
+                    completion(nil)
                 } else {
-                    completion(false)
+                    print("Pre-Chat Lead Collection is not enabled.")
+                    
+                    let applozicClient = applozicClientType.init(applicationKey: appID)
+                    let kmUser = KMUser()
+                    kmUser.userId = Kommunicate.randomId()
+                    kmUser.applicationId = applicationId
+                    
+                    Kommunicate.registerUser(kmUser, completion: {
```

Comment:
We have the same code below in prechat delegate to register and create conversation, we can reuse it.